packages/flutter_tools/test/general.shard/android/android_device_test.dart

```diff
@@ -637,6 +637,67 @@ flutter:
       ProcessManager: () => mockProcessManager,
     });
   });
+
+  test('Can parse adb shell dumpsys info', () {
+    const String exampleOutput = r'''
+Applications Memory Usage (in Kilobytes):
+Uptime: 441088659 Realtime: 521464097
+
+** MEMINFO in pid 16141 [io.flutter.demo.gallery] **
+                   Pss  Private  Private  SwapPss     Heap     Heap     Heap
+                 Total    Dirty    Clean    Dirty     Size    Alloc     Free
+                ------   ------   ------   ------   ------   ------   ------
+  Native Heap     8648     8620        0       16    20480    12403     8076
+  Dalvik Heap      547      424       40       18     2628     1092     1536
+ Dalvik Other      464      464        0        0
+        Stack      496      496        0        0
+       Ashmem        2        0        0        0
+      Gfx dev      212      204        0        0
+    Other dev       48        0       48        0
+     .so mmap    10770      708     9372       25
+    .apk mmap      240        0        0        0
+    .ttf mmap       35        0       32        0
+    .dex mmap     2205        4     1172        0
+    .oat mmap       64        0        0        0
+    .art mmap     4228     3848       24        2
+   Other mmap    20713        4    20704        0
+    GL mtrack     2380     2380        0        0
+      Unknown    43971    43968        0        1
+        TOTAL    95085    61120    31392       62    23108    13495     9612
+
+ App Summary
```

Comment:
This section is missing from the documentation. Do different Android versions output different info?

Reply:
Added to documentation. Not that I know of

Reply:
Limiting the parsing to just the 'App Summary' section might help with different versions.  The only differences I seen have to do with older Android version 7 I think.

Older (Android 7 I think)

`  Dalvik Heap   5110(3)    0    4136    4988(3)    0       0    9168    8958(6)  210
`

This is the modern output for a few years:
 
`  Dalvik Heap    34340    33816        0        0    62436    53883     8553
`

But I only care about the App Summary section.